centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -3112,3 +3113,276 @@ function sanitizeFormHostParameters(array $ret): array
     }
     return $bindParams;
 }
+
+/**
+ * Create a new host from formData.
+ *
+ * @param array<mixed> $ret
+ *
+ * @return int|null
+ */
+function insertHostInAPI(array $ret = []): int|null
+{
+    global $centreon, $form, $isCloudPlatform, $basePath;
+
+    /** @var array<string,int|string|null> $formData */
+    $formData = !count($ret) ? $form->getSubmitValues() : $ret;
+
+    try {
+        $hostId = insertHostByApi($formData, $isCloudPlatform, $basePath);
+
+        if ((int) $formData['host_register'] === 0) {
+            updateHostTemplateService($hostId);
+        }
+
+        if (! $isCloudPlatform) {
+            if ((int) $formData['host_register'] !== 0) {
+                updateHostHostParent($hostId, $formData);
+                updateHostHostChild($hostId);
+            }
+            updateHostContactGroup($hostId, $formData);
+            updateHostContact($hostId, $formData);
+        }
+
+        if (
+            ! empty($formData['dupSvTplAssoc']['dupSvTplAssoc'])
+            || $isCloudPlatform === true
+        ) {
+            createHostTemplateService($hostId);
+        }
```

Comment:
Those methods does not have equivalent in API ?

Reply:
No at the moment no